src/main/kotlin/cacophony/controlflow/CFGNode.kt

```diff
@@ -0,0 +1,9 @@
+package cacophony.controlflow
+
+sealed interface CFGRootNode
+
+sealed interface CFGLeafNode
+
+sealed class CFGNode {
+    // TODO
+}
```

Comment:
**suggestion:** In my PR I opted for a `CFGVertex` class that contains a `CFGNode` to represent a vertex with its neighbors. I think you can just rename `CFGRootNode` to `CFGVertex` here and I can then resolve any conflicts once this is merged (we can also go for `CFGRootNode` but I think `CFGVertex` is less confusing.

Reply:
I agree. I just wanted some superclass for all roots and leaves. I will also delete `CFGLeafNode` as you have `Leaf` in your PR.